### DISCUSSION TITLE
Migrating From Separate Enterprise Servers

### DISCUSSION THREAD

**Discussion by jfarrell423** [github/gh-gei#1234] [Q&A]
How do I migrate my repos from an organization from an Enterprise github.myprivateserver.com to Enterprise github.com.
In the process I want to transfer Issues, Pull Requests, Projects, Wiki, and Insights. Both locations are on separate enterprise servers.

Been trying in a power shell script using 
$env:GH_SOURCE_PAT          =<token>
$env:GH_PAT                        =<token>
$env:GH_HOST$                   =<host name>
env:GH_ENTERPRISE_TOKEN  =<token> 

gh gei migrate-repo --github-source-org <org> --source-repo <private enterprise runner repo> --github-target-org <org> --target-repo <repo name>

Upper management wants to keep as much of the repo history as possible. I would appreciate any help someone can give me.
Would like to know if this type of migration is even possible. I am getting the feeling that both repos have to be on the same server.

**Comment by dylan-smith** [comment]
GEI definitely supports migrating repos from GH Enterprise Server to github.com. The docs on this can be found here: https://docs.github.com/en/migrations/using-github-enterprise-importer/migrating-between-github-products/migrating-repositories-from-github-enterprise-server-to-github-enterprise-cloud

**Reply by jfarrell423** [reply]
Sorry I should have put my reply here, I just posed a new message.

**Reply by jfarrell423** [reply]
We are using VM Ware to host our private enterprise server. We are trying to migrate to github.com Enterprise server.

**Reply by brianaj** [reply]
@jfarrell423 gei does not support migrating to another GHES instance. You would need to use `ghe-migrator` following these [docs](https://docs.github.com/en/migrations/using-ghe-migrator/about-ghe-migrator). 

_Note: github.com != Enterprise server_

**Reply by jfarrell423** [reply]
The license we have we log into GIHUB.COM but it is an enterprise account.

**Reply by jfarrell423** [reply]
https://github.com/enterprises/UTENN-EMU/sso

**Reply by brianaj** [reply]
@jfarrell423 Got it, then you are migrating into github.com not Enterprise Server. @dylan-smith already linked you to the proper [docs](https://docs.github.com/en/migrations/using-github-enterprise-importer/migrating-between-github-products/migrating-repositories-from-github-enterprise-server-to-github-enterprise-cloud#migrate-a-single-repository). So you should be using `--ghes-api-url` to answer your question below. For hands on support you can navigate through the `Contact Us` link and our customer support squad can help you out further.

**Comment by jfarrell423** [comment]
How do I tell the tool to pull from server a and push to server b? My admin has given me owner on both servers, but when it comes to server a, I get the message that it cannot find the organization. So our local Enterprise server is hosted on our private vm. I have firewall access to both locations. So if I am pulling an individual repo should I be setting these variables for my private location? AWS_REGION, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS_SESSION_TOKEN. Or should I be using --ghes-api-url GHES-API-URL and just generate a script cutting the repos out in chunks for test runs.
